src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java

```diff
@@ -0,0 +1,116 @@
+package com.ss.ita.rozetka.pageobject.elements;
+
+import io.qameta.allure.Step;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.openqa.selenium.support.Color;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.codeborne.selenide.Selenide.*;
+
+@RequiredArgsConstructor
+public class Product {
+
+    @NonNull
+    private final String xPath;
```

Comment:
why is there a need to pass xpath in constructor?

Reply:
each Product on the ProductTypePage has its xpath (difference by index in the elements collection $$x(//div[@class='goods-tile__inner').
when I want to get Product by index from the collection, I use
public Product getProduct(int numberProduct) {
    return new Product(String.format(("(//div[@class='goods-tile__inner'])[%s]"), numberProduct))
} from ProductTypePage
xpath from Product uses in the get* methods for locate product elements(such as title, price....)